command/job_stop.go

```diff
@@ -140,92 +140,147 @@ func (c *JobStopCommand) Run(args []string) int {
 		return 1
 	}
 
-	// Check if the job exists
-	jobs, _, err := client.Jobs().PrefixList(jobID)
-	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Error deregistering job: %s", err))
-		return 1
-	}
-	if len(jobs) == 0 {
-		c.Ui.Error(fmt.Sprintf("No job(s) with prefix or id %q found", jobID))
-		return 1
+	statusCh := make(chan int, len(jobIDs))
+
+	var wg sync.WaitGroup
+	for _, jobID := range jobIDs {
+		jobID := jobID
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			// Truncate the id unless full length is requested
+			length := shortId
+			if verbose {
+				length = fullId
+			}
+
+			// Check if the job exists
+			jobs, _, err := client.Jobs().PrefixList(jobID)
+			if err != nil {
+				// TODO: can we improve this error message?
+				c.Ui.Error(fmt.Sprintf("Error finding jobs with prefix: %s err: %s", jobID, err))
+				statusCh <- 1
+				return
+			}
+			if len(jobs) == 0 {
+				c.Ui.Error(fmt.Sprintf("No job(s) with prefix or id %q found", jobID))
+				statusCh <- 1
+				return
+			}
+			if len(jobs) > 1 {
+				if (jobID != jobs[0].ID) || (c.allNamespaces() && jobs[0].ID == jobs[1].ID) {
+					c.Ui.Error(fmt.Sprintf("Prefix matched multiple jobs\n\n%s", createStatusListOutput(jobs, c.allNamespaces())))
+					statusCh <- 1
+					return
+				}
+			}
+
+			// Prefix lookup matched a single job
+			q := &api.QueryOptions{Namespace: jobs[0].JobSummary.Namespace}
+			job, _, err := client.Jobs().Info(jobs[0].ID, q)
+			if err != nil {
+				c.Ui.Error(fmt.Sprintf("Error deregistering job with id %s err: %s", jobID, err))
+				statusCh <- 1
+				return
+			}
+
+			getConfirmation := func(question string) (int, bool) {
+				answer, err := c.Ui.Ask(question)
+				if err != nil {
+					c.Ui.Error(fmt.Sprintf("Failed to parse answer: %v", err))
+					return 1, false
+				}
+
+				if answer == "" || strings.ToLower(answer)[0] == 'n' {
+					// No case
+					c.Ui.Output("Cancelling job stop")
+					return 0, false
+				} else if strings.ToLower(answer)[0] == 'y' && len(answer) > 1 {
+					// Non exact match yes
+					c.Ui.Output("For confirmation, an exact ‘y’ is required.")
+					return 0, false
+				} else if answer != "y" {
+					c.Ui.Output("No confirmation detected. For confirmation, an exact 'y' is required.")
+					return 1, false
+				}
+				return 0, true
+			}
+
+			// Confirm the stop if the job was a prefix match
+			// Ask for confirmation only when there's just one
+			// job that needs to be stopped. Since we're stopping
+			// jobs concurrently, we're going to skip confirmation
+			// for when multiple jobs need to be stopped.
+			if len(jobIDs) == 1 && jobID != *job.ID && !autoYes {
+				question := fmt.Sprintf("Are you sure you want to stop job %q? [y/N]", *job.ID)
+				code, confirmed := getConfirmation(question)
+				if !confirmed {
+					statusCh <- code
+					return
+				}
+			}
+
+			// Confirm we want to stop only a single region of a multiregion job
+			if len(jobIDs) == 1 && job.IsMultiregion() && !global {
```

Comment:
For this specific check, do you think this behavior looks good i.e. just confirming only for when just a single job is passed as an argument? I was thinking, and for the record, I'm not completely clear on the use case for this flow, that it _looks_ like something people would be interested in even with multiple jobs, do you think it could be another flag? Keeping in mind we don't introduce flag creep though.

Reply:
Ah thanks for the callout on this, definitely subtle. +1 to your approach though.

I think it follows the general logic of "if users are passing in multiple job ids we skip any interactive prompts."

Reply:
Thanks for the response, just to confirm, so you are saying we can have another flag to incorporate this behavior when we have multiple jobs?

Reply:
I just noticed this line should check `autoYes` like the like previous `if` statement!

> so you are saying we can have another flag to incorporate this behavior when we have multiple jobs?

I don't think we need another flag right now. A good UX for concurrent confirmations seems difficult, and I don't think we need to slow down this effort to implement it. So for this code just follow what we do in the previous `if` statement: only prompt if there's only 1 job being stopped.